### PR TITLE
Add flash driver and blocking modulefor W91

### PR DESCRIPTION
**Pull request by serhiiSalamakha**
http://192.168.48.49:8080/browse/ZEPHYR-433
http://192.168.48.49:8080/browse/ZEPHYR-505

### CODE REVIEW

**Comment by andriy-bilynskyy** [comment]
Please change commit message: "modulefor" -> "module for"

File: soc/riscv/riscv-privilege/telink_w91/idle.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_w91/idle.c]: Why do we add this file? Are there any differences from standard risc-v implementation?

File: soc/riscv/riscv-privilege/telink_w91/compiler.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/*
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_w91/compiler.h]: This file seems like duplication of standard zephyr functionality
See: https://docs.zephyrproject.org/apidoc/latest/gcc_8h_source.html

File: soc/riscv/riscv-privilege/telink_w91/blocking_core/blocking.c [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+	uint32_t steps_timeout = CONFIG_BLOCKING_CORE_TELINK_W91_CORE_STOP_TIMEOUT_US /
+			CONFIG_BLOCKING_CORE_TELINK_W91_CHECK_STATE_PERIOD_US;
+	uint32_t steps_count = 0;
+
+	key = irq_lock();
+
+	atomic_set(&blocking_state, BLOCKING_CORE_STOPPED_STATE);
+
+	while (atomic_get(&blocking_state) != BLOCKING_CORE_ACTIVE_REQ_STATE) {
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_w91/blocking_core/blocking.c:88]: Seems we left this cycle by timeout always...

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

File: soc/riscv/riscv-privilege/telink_w91/idle.c (hunk 1 above)
Comment (serhiiSalamakha) [reply soc/riscv/riscv-privilege/telink_w91/idle.c]: removed

File: soc/riscv/riscv-privilege/telink_w91/compiler.h (hunk 2 above)
Comment (serhiiSalamakha) [reply soc/riscv/riscv-privilege/telink_w91/compiler.h]: removed

File: soc/riscv/riscv-privilege/telink_w91/blocking_core/blocking.c (hunk 3 above)
Comment (serhiiSalamakha) [reply soc/riscv/riscv-privilege/telink_w91/blocking_core/blocking.c:88]: The N22 core sets BLOCKING_CORE_ACTIVE_REQ_STATE

**Pull request closed without merging**